KEDamage.cs

```diff
@@ -8,58 +8,94 @@ class KEDamage:Damage
     {
         public struct KineticData
         {
+            /// <summary>
+            /// The pierce of the shot
+            /// </summary>
             public float Pierce;
-
+            /// <summary>
+            /// Distance between the firing unit and target unit
+            /// </summary>
             public float Distance;
-
-            public float ArmorKEFactor;
-
-            public float HealthKEFactor;
-
+            /// <summary>
+            /// Multiplier for armor degradation
+            /// </summary>
+            public float Degradation;
+            /// <summary>
+            /// Multiplier for health damage
+            /// </summary>
+            public float HealthDamageFactor;
+            /// <summary>
+            /// Air friction constant used in calculation of attenuation
+            /// </summary>
             public float FrictionFactor;
         }
 
-        private KineticData data;
-        private Damage.Target target;
+        private KineticData _keData;
 
-        public KEDamage(KineticData d,Damage.Target t):base(EDamageTypes.KE)
+        public KEDamage(KineticData data,Target target)
```

Comment:
I wouldn't recommend `KEDamage` instead of `KineticDamage`. That's up to you. 

Also: `public KEDamage(KineticData data,Target target)` needs spacing -> `public KEDamage(KineticData data, Target target)`.